pkg/controllers/state/cluster.go

```diff
@@ -416,6 +448,7 @@ func (c *Cluster) ClearPodSchedulingMappings(podKey types.NamespacedName) {
 	c.podAcks.Delete(podKey)
 	c.podsSchedulableTimes.Delete(podKey)
 	c.podsSchedulingAttempted.Delete(podKey)
+	c.podHealthyNodePoolScheduledTime.Delete(podKey)
```

Comment:
This looks to me that we are keeping this in our cluster state forever if the pod isn't deleted -- I think we should throw this element away once the pod goes ready -- are we handling that correctly in the metric controller?

Reply:
We flush this element here - https://github.com/kubernetes-sigs/karpenter/blob/main/pkg/controllers/metrics/pod/controller.go#L329